Add De Lijn component documentation

bollewolle opened the pull request:
**Description:**
Documentation of the De Lijn component to add passages of stops of the Flemish public transport company into home assistant

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** https://github.com/home-assistant/home-assistant/pull/24265

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

bollewolle committed:
Add De Lijn component

Documentation of the De Lijn component to add passages of stops of the Flemish public transport company into home assistant

bollewolle committed:
Merge pull request #1 from bollewolle/delijn

Add De Lijn component

klaasnicolaas commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,76 @@
+comments: false
+sharing: true
+footer: true
+ha_release: "0.94"
```
```suggestion
ha_release: 0.94
```

klaasnicolaas commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,76 @@
+sharing: true
+footer: true
+ha_release: "0.94"
+ha_category: 
```
it's also a `sensor` 😉

bollewolle replied:
Makes sense :) Thanks for the suggestions

klaasnicolaas commented on source/_components/delijn.markdown:
````diff
@@ -0,0 +1,76 @@
+
+```yaml
+# Example configuration.yaml entry
+sensor:
````
Only required items in the first example

bollewolle replied:
Thanks, adjusted in a new commit.

klaasnicolaas committed:
:pencil2: Tweak

bollewolle committed:
Update source/_components/delijn.markdown

Suggested corrections applied

Co-Authored-By: Klaas Schoute <klaas_schoute@hotmail.com>

bollewolle committed:
Some suggested changes added

Sensor added as ha_category.
Only show the mandatory items in the configuration example.
Rewritten the Configuration part a little bit to align with example.
Added some additional information on the Full Configuration to clarify the outcome of that configuration.

bollewolle commented:
Made some additional changes as suggested and some additional clarifications.

klaasnicolaas commented on source/_components/delijn.markdown:
````diff
@@ -0,0 +1,75 @@
+```
+
+{% configuration %}
+subscriptionkey:
````
Shouldn't this also be `sub_key`? At least that's how I see it in your code.

bollewolle replied:
Absolutely correct, noticed another incorrect name as well so corrected that one as well.

bollewolle committed:
Alignment of configuration names

Aligned the names in the configuration explanation with the actual configuration names.

bollewolle committed:
Linked to custom lovelace card added

Added link to custom lovelace card that makes best use of all the retrieved features (ie. correct colour codes of the line cards)

frenck commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,78 @@
+  - Transport
+  - Sensor
+ha_iot_class: Cloud Polling
+ha_qa_scale: 
```
If the integration has no scale, you should remove this line.

bollewolle replied:
Indeed, reference is removed

frenck commented on source/_components/delijn.markdown:
````diff
@@ -0,0 +1,78 @@
+      max_passages: 20
+```
+
+## {% linkable_title Custom lovelace card %}
````
Extra new line:
```suggestion
## {% linkable_title Custom Lovelace card %}

```

frenck commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,78 @@
+ha_qa_scale: 
+---
+
+The `delijn` sensor will give you the departure time of the next bus, tram or subway a specifi stop of the De Lijn public transport network in Flanders (Belgium).
```
```suggestion
The `delijn` sensor will give you the departure time of the next bus, tram or subway a specific stop of the De Lijn public transport network in Flanders (Belgium).
```

frenck commented on source/_components/delijn.markdown:
````diff
@@ -0,0 +1,78 @@
+```
+
+## {% linkable_title Custom lovelace card %}
+Works best with the following custom lovelace card: https://github.com/bollewolle/delijn-card
````
```suggestion
Works best with the following custom Lovelace card: https://github.com/bollewolle/delijn-card
```

frenck commented:
Hi there @bollewolle, I've reviewed as well and left some comments. Could you please take a look? 👍

klaasnicolaas commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,78 @@
+comments: false
+sharing: true
+footer: true
+ha_release: 0.94
```
```suggestion
ha_release: 0.96
```

or 0.95 but the cut-off will tonight or tomorrow.

bollewolle committed:
Apply suggestions from code review

Co-Authored-By: Klaas Schoute <klaas_schoute@hotmail.com>
Co-Authored-By: Franck Nijhof <frenck@frenck.nl>

bollewolle committed:
Documents aligned with changes in component code

sub_key was replaces with api_key in the component code so the reference is changed in the documentation here as well.

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

bollewolle committed:
Updated with changes to component

The delijn component has been updated based on feedback given in PR #24265. Based on those changes the documentation had to be updated as well. Now documentation and component are aligned again.

frenck committed:
:pencil2: Markdown cleanup

MartinHjelmare commented on source/_components/delijn.markdown:
```diff
@@ -0,0 +1,72 @@
+  - Transport
+  - Sensor
+ha_iot_class: Cloud Polling
+---
```
Should we add a logo?

klaasnicolaas replied:
yes we should 😄

bollewolle replied:
Can I add it via github directly? I don't have a local version of the repo on my pc and I think I read you could not add a picture via github. If possible, please let me know how. Otherwise, if you could add them yourself, these are some that I found:
- the logo on their site: https://static.delijn.be/Images/dl-logo_tcm3-1040.svg
- the logo on Wikipedia: https://nl.wikipedia.org/wiki/De_Lijn_(vervoermaatschappij)#/media/Bestand:De_Lijn.svg
- a vector version of their logo (although the blue seems a little bit off): https://seeklogo.com/vector-logo/39175/de-lijn

Hope it helps. Cheers!

frenck replied:
Done.

frenck committed:
:art: Adds De Lijn logo
